pkg/runtime/unstructured.go

```diff
@@ -28,7 +28,9 @@ import (
 // type, which can be used for generic access to objects without a predefined scheme.
 var UnstructuredJSONScheme ObjectDecoder = unstructuredJSONScheme{}
 
-type unstructuredJSONScheme struct{}
+type unstructuredJSONScheme struct {
+	Decoder
```

Comment:
It either needs an embedded Decoder or a dummy DecodeIntoWithDefaultVersionKind()